### PR TITLE
Improves multicluster documentation

### PR DESCRIPTION
**Pull request by sdake**
Improve documentation based upon fresh eyes running through the
documented process.

### CODE REVIEW

File: _docs/setup/kubernetes/multicluster-install.md [hunk 1]
```diff
@@ -106,12 +106,27 @@ on **one** Kubernetes cluster.
...
 The istio-remote component must be deployed to each remote Kubernetes cluster.
 
+### Set environment variables for Pod IPs from Istio control plane needed by remote
+
+> Please wait for the Istio control plane to finish initializing before proceeding to steps
+in this section.
+
+```bash
+export PILOT_POD_IP=$(kubectl -n istio-system get pod -l istio=pilot -o jsonpath='{.items[0].status.podIP}')
+export POLICY_POD_IP=$(kubectl -n istio-system get pod -l istio=mixer -o jsonpath='{.items[0].status.podIP}')
```
Comment (tiswanso) [inline _docs/setup/kubernetes/multicluster-install.md]: I haven't been able to test these commands to validate the correct pod is found in items[0] all the time.  I could try later this evening / tomorrow a.m. EST.  I believe the policy, telemetry, and statsd-prom-bridge pods all get the mixer label.

**Comment by andraxylia** [comment]
What is "nodes' pod CIDR", the pod cidr or the node cidr? I have met "nodes' pod" before and find it confusing.

**Comment by andraxylia** [comment]
Link to the "Istio control plane" is broken.

**Comment by andraxylia** [comment]
Warning link is also broken.

**Comment by andraxylia** [comment]
Please remove or rephrase "Use your desired technique to deploy all Kubernetes clusters that are to participate in the mesh with only Kubernetes and the CNI of choice. " - Istio has pre-requisite Kubernetes cluster and the choice of CNI belong in the k8s documentation, we do not need to get into that and confuse the user. Use the simple setup as a reference. Also, please qualify all acronyms and add links to relevant details (CIDR, CNI etc).

**Comment by andraxylia** [comment]
Overall comment: it's not clear where the command are run, on the master vs the remote cluster.

File: _docs/setup/kubernetes/multicluster-install.md [hunk 2]
```diff
@@ -106,12 +106,27 @@ on **one** Kubernetes cluster.
...
+with Tiller on each remote, take care to copy the environment variables to each node before running
+`helm install`.
+
 ### Use kubectl with Helm to connect the remote cluster to the local
 
 1. Use the helm template command on a remote to specify the Istio control plane service endpoints:
 
    ```bash
-   helm template install/kubernetes/helm/istio-remote --name istio-remote --set pilotEndpoint=`pod_ip_of_pilot_local` --set policyEndpoint=`pod_ip_of_policy_local` --set statsdEndpoint=`pod_ip_of_statsd_local` > $HOME/istio-remote.yaml
+   helm template install/kubernetes/helm/istio-remote --name istio-remote --set global.pilotEndpoint=${PILOT_POD_IP} --set global.policyEndpoint=$(POLICY_POD_IP} --set global.statsdEndpoint=${STATSD_POD_IP} > $HOME/istio-remote.yaml
```
Comment (andraxylia) [inline _docs/setup/kubernetes/multicluster-install.md]: People usually read in sequence, so the fact there are two alternative for installation has to be specified before "Use kubectl with Helm ..."

File: _docs/setup/kubernetes/multicluster-install.md [hunk 3]
```diff
@@ -138,7 +153,7 @@ install one:
 1. Install the Helm chart:
 
    ```bash
-   helm install install/kubernetes/helm/istio-remote --name istio-remote --set pilotEndpoint=`pod_ip_of_pilot_local` --set policyEndpoint=`pod_ip_of_policy_local` --set statsdEndpoint=`pod_ip_of_statsd_local` > $HOME/istio-remote.yaml -n istio-system
+   helm install install/kubernetes/helm/istio-remote --name istio-remote --set global.pilotEndpoint=${PILOT_POD_IP} --set global.policyEndpoint=${POLICY_POD_IP} --set global.statsdEndpoint=${STATSD_POD_IP} --namespace istio-system
    ```
 
 ### Helm configuration parameters
```
Comment (andraxylia) [inline _docs/setup/kubernetes/multicluster-install.md]: Are the config params needed for both alternative or just the last one? What is the user supposed to do with these config, define them or what? Shouldn't this be before any helm command is executed?

**Comment by andraxylia** [comment]
General comment: the doc is missing to explain in the beginning that one k8s cluster is used as master and runs the istio control plane, and the other "remote" clusters are slaves and run just istio-remote. It would also be good to specify what istio-remote purpose is.

**Comment by andraxylia** [comment]
"Create a namespace for instantiating the secrets:" keep in mind that for general installation istio-system is not hardcoded and the users has the option of changing the name. Please specify this is the same namespace that will be used for istio installation.

**Comment by andraxylia** [comment]
Please move "warnings" as "notes" in the beginning of each step and think the users will algorithmically execute your steps. Do not assume people read the entirety of steps before executing (if you need proof of that, I have a youtube video with somebody following Istio installation steps without any pre-read).

File: _docs/setup/kubernetes/multicluster-install.md (hunk 1 above)
Comment (sdake) [reply _docs/setup/kubernetes/multicluster-install.md]: policy/telemetry use mixer: https://github.com/istio/istio/blob/master/install/kubernetes/helm/istio/charts/mixer/templates/deployment.yaml#L136 and https://github.com/istio/istio/blob/master/install/kubernetes/helm/istio/charts/mixer/templates/deployment.yaml#L136

statsd-prom-bridge uses https://github.com/istio/istio/blob/master/install/kubernetes/helm/istio/charts/mixer/templates/statsdtoprom.yaml#L2-L12

telemetry currently isn't managed by istio-remote.  I'm not clear if it should be or not.

**Comment by sdake** [comment]
@gyliu513 What I do is hack apart bookinfo into two parts:

Part 1 has everything except reviews-v3.  I deploy this on one cluster.

Then I deploy reviews-v3 and ratings on the remote cluster.  Once this is done, you should see red stars enter the rotation of the bookinfo app.  To verify that your remote cluster is actually serving traffic, you can force it via the routerules-v3 rule.  I have uploaded my demo apps (you will have to change the proxy source/tag, or just use mine) so you can give it a spin.

The two files are:
bookinfo-injected-notls.yaml (deploy this on the node that has the ingress controller)
remote-app.yaml (deploy this on the node that you deployed the remote to).
route-rule-reviews-v3.yaml (apply this route rule, you should see 100% red stars afterwards)

This will show that two kubernetes control planes are running a single mesh.

I did intend to already have this bookinfo sample already documented for 0.8, but just not there yet.

Make sure to deploy without TLS (or you may need to enable some of the cacert works in deploy.sh.

[demo2.tar.gz](https://github.com/istio/istio.github.io/files/1967652/demo2.tar.gz)

**Comment by gyliu513** [comment]
Thanks @sdake 

Is there a feature list for istio mc support? Does istio mc support 

- Traffic split for different clusters?
- Topology overview for applications in different clusters?
- More from your point?

**Comment by sdake** [comment]
Feature list for Istio MC support:
I don't think anyone has written down a feature list publicly, at-least that I am aware of.

Split traffic for different clusters?  I don't understand what you mean here.
By topology overview, do you mean some form of GUI to show how applications connect?
More from which point?

The original use case we were working towards is as follows:
Execute a blue/green (or canary) deployment on an entirely new Kubernetes cluster.  The canary deployment starts out with a new version of one component of the software (e.g. reviews-v3).  Route rule is set to 5% to go to canary deployment.  If after several hours, the operator is satisfied, they can up the route rule to 20%.  If after several hours, operator still satisfied, they can go to 50%.  After a few more hours, things looking good under load, 100% of traffic is forwarded.

The ability to use this with multiple kubernetes clusters is simply the beginning of the journey.  The other tech was already there.

**Comment by gyliu513** [comment]
Thanks @sdake 

- Traffic split for different clusters. This is same as you mentioned for blue/green (or canary) deployment, suppose I have two clusters, I can deploy applications to two clusters, and then set route rules to split the traffic for the applications cross clusters? Such as 70% to version-1 in cluster1 and 30% to version-2 in cluster2?

- Topology overview means I want to view the dependencies of the whole application cross clusters, does the istio mc support such case?

**Comment by gyliu513** [comment]
@sdake did some test for your demo cases, but found it is not working in my cluster, the request never goes to `reviews-v3`, and after I create the route rules as follows, product page info report error.

```
apiVersion: config.istio.io/v1alpha2
kind: RouteRule
metadata:
  name: reviews-default
spec:
  destination:
    name: reviews
  precedence: 1
  route:
  - labels:
      version: v3
    weight: 100
```

image

serivcegraph has no info for `reviews-v3`
image

Any suggestions for me to debug this issue?

**Comment by john-a-joyce** [comment]
@gyliu513 - All the Istio route-rule features should work in multicluster mode.  There is nothing in the design that precludes the features working.  At the current time e2e testing is still in progress to ensure everything works and over time I hope that we could extend the tests to include specific uses cases for multicluster.   If you have such a use case let us know, raise an issue or extent the tests directly on your own.  It is likely we will find some gaps as people push the envelope on this. 

Regarding topology overview - The multicluster changes don't provide any additional capabilities over Istio in a single cluster mode.   You can you use tools like service graph etc. to view the topology.  If everything is set right they should provide a view across all clusters.

**Comment by john-a-joyce** [comment]
@gyliu513 - A few things you should do.   
1.  Check the pilot logs to confirm review-v3 is able to connect to pilot and doing an EDS requests  should see something like:  2018-05-02T20:17:24.293804Z     info    EDS: ACK sidecar~192.170.228.103~reviews-v3-7d9bc85dcb-jm5xt.istio-system~istio-system.svc.cluster.local-1030 2018-05-02 20:17:24.077780316 +0000 UTC m=+1371.308437668 [prometheus.istio-system.svc.cluster.local|http-prometheus]
2. Check the pilot xDS cache (e.g. curl the registration) and confirm there is an endpoint address for the reviews-v3 pod
3.  Go into the productpage and Reviews-v3 pods (sidecar container) and ensure they can ping each other.

**Comment by tiswanso** [comment]
Hi @gyliu513 -- did you performed the CA-cert steps described here on both clusters? https://istio.io/docs/tasks/security/plugin-ca-cert.html 

Possibly a better forum for getting help is the istio slack channel.  Also, since this PR has merged, creating an issue to continue this discussion is the best way forward.

**Comment by gyliu513** [comment]
Thanks @john-a-joyce  and @tiswanso 

@john-a-joyce

> Check the pilot logs to confirm review-v3 is able to connect to pilot and doing an EDS requests should see something like:

No log related to `reviews-v3`, but all of the logs are for `reviews-v1` and `reviews-v2`

> Check the pilot xDS cache (e.g. curl the registration) and confirm there is an endpoint address for the reviews-v3 pod

How to curl the registration? Can you please show me a command? I do not how to get the registration URL.

> Go into the productpage and Reviews-v3 pods (sidecar container) and ensure they can ping each other.

Yes, they can ping each other cross cluster.

@tiswanso 

> did you performed the CA-cert steps described here on both clusters? https://istio.io/docs/tasks/security/plugin-ca-cert.html

I was using `istio.yaml` to deploy istio, so seems no need to do?

**Comment by gyliu513** [comment]
@sake @john-a-joyce @tiswanso A new issue is opened at https://github.com/istio/istio.github.io/issues/1235

**Pull request merged**